job/service.go

```diff
@@ -303,6 +307,46 @@ func (srv *Service) Check(ctx context.Context, namespace models.NamespaceSpec, j
 	return err
 }
 
+func (srv *Service) GetJobBasicInfo(ctx context.Context, jobSpec models.JobSpec) models.JobBasicInfo {
+	var jobBasicInfo models.JobBasicInfo
+	jobBasicInfo.Spec = jobSpec
+
+	dest, err := srv.pluginService.GenerateDestination(ctx, jobSpec, jobSpec.NamespaceSpec)
+	if err != nil {
+		jobBasicInfo.Log.Write(writer.LogLevelError, fmt.Sprintf("unable to generate destination, err:%v", err))
+	}
+	if dest != nil {
+		destination := models.JobSpecTaskDestination{
+			Destination: dest.Destination,
+			Type:        dest.Type,
+		}
+		jobBasicInfo.Destination = destination.URN()
+	}
+
+	deps, err := srv.pluginService.GenerateDependencies(ctx, jobSpec, jobSpec.NamespaceSpec, false)
+	if err != nil {
+		jobBasicInfo.Log.Write(writer.LogLevelError, fmt.Sprintf("failed to generate dependencies, err:%v", err))
+	} else {
+		if deps != nil {
+			jobBasicInfo.JobSource = deps.Dependencies
+		} else {
+			jobBasicInfo.Log.Write(writer.LogLevelWarning, "no dependencies detected")
```

Comment:
should we change the error message to `no job sources detected`? not finding anything does not mean there are no dependencies for this job, there might be static dependencies.

Reply:
we might also need to change `failed to generate dependencies` to `failed to generate job sources`, to avoid confusion with job dependency/upstream section later